janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/NumericTranslationQueryParser.java

```diff
@@ -0,0 +1,149 @@
+    }
+
+    @Override
+    protected Query newTermQuery(Term t) {
+        if (t.field() == null) {
+            return super.newTermQuery(t);
+        }
+
+        Class<?> dataType = storeRetriever.get(t.field()).getDataType();
+        if (Number.class.isAssignableFrom(dataType)) {
+            try {
+                return buildNumericQuery(t.field(), t.text(), dataType);
+            } catch (NumberFormatException e) {
+                log.warn("Expected Number type for " + t.field() + ":" + dataType + ", but " + e.getMessage());
+            }
+        }
+
+        return super.newTermQuery(t);
+    }
+
+    @Override
+    protected Query newWildcardQuery(final Term t) {
+        if (t.field() == null) {
+            return super.newTermQuery(t);
+        }
+
+        Class<?> dataType = storeRetriever.get(t.field()).getDataType();
+        if (Number.class.isAssignableFrom(dataType)) {
+            try {
+                return buildNumericQuery(t.field(), t.text(), dataType);
+            } catch (NumberFormatException e) {
+                log.warn("Expected Number type for " + t.field() + ":" + dataType + ", but " + e.getMessage());
+            }
+        }
+
+        return super.newWildcardQuery(t);
+    }
+
+    private Query getNumericRangeQuery(final String field, final Class<?> type, String start, String end, final boolean includeLower,
+                                       final boolean includeUpper) {
+        if (AttributeUtil.isWholeNumber(type)) {
+            Long min = isMatchAll(start) ? Long.MIN_VALUE : Long.parseLong(start);
+            Long max = isMatchAll(end) ? Long.MAX_VALUE : Long.parseLong(end);
+            if (!includeLower) {
+                min = Math.addExact(min, 1);
+            }
+            if (!includeUpper) {
+                max = Math.addExact(max, -1);
+            }
+            return LongPoint.newRangeQuery(field, min, max);
+        } else {
+            Double min = isMatchAll(start) ? Double.NEGATIVE_INFINITY : Double.parseDouble(start);
+            Double max = isMatchAll(end) ? Double.POSITIVE_INFINITY : Double.parseDouble(end);
+            if (!includeLower) {
+                min = DoublePoint.nextUp(min);
+            }
+            if (!includeUpper) {
+                max = DoublePoint.nextDown(max);
+            }
+            return DoublePoint.newRangeQuery(field, min, max);
+        }
+    }
+
+    private Query buildNumericQuery(final String field, final String value, Class<?> type) {
+        if (AttributeUtil.isWholeNumber(type)) {
+            if (isMatchAll(value)) {
+                return LongPoint.newRangeQuery(field, Long.MIN_VALUE, Long.MAX_VALUE);
+            }
+            return LongPoint.newExactQuery(field, Long.parseLong(value));
+        } else {
+            if (isMatchAll(value)) {
+                DoublePoint.newRangeQuery(field, Double.NEGATIVE_INFINITY, Double.POSITIVE_INFINITY);
```

Comment:
I am not sure but possibly you missed `return` here

Reply:
Yes, you right